broadcast/broadcast-client/client_builder.go

```diff
@@ -12,22 +12,26 @@ type builder struct {
 	client    httpclient.HTTPInterface
 }
 
+// Builder is a builder func for the broadcast client.
 func Builder() *builder {
 	return &builder{}
 }
 
+// WithHttpClient sets the http client to be used by the broadcast client. It requires a httpclient.HTTPInterface to be passed.
```

Comment:
Nope Damian, if someone doesn't pass the http client it will be created here:
![image](https://github.com/bitcoin-sv/go-broadcast-client/assets/48433067/317fba03-ce33-4348-a10f-063d83cc1121)